### ISSUE TITLE
[FEATURE]: data truncation in Colossal-LLaMA-2

### ISSUE THREAD

**Issue by scikkk** [hpcaitech/ColossalAI#4968]
### Describe the feature

Sometimes simple truncation may lead to the loss of a large amount of data. Is it possible to add an option that can split the data instead of truncating it?

**Comment by Orion-Zheng** [comment]
Good point! 😃The current spliced_and_tokenized_dataset only support concatenating multiple samples into one and will truncate extra tokens. It's probably because most data points in pre-training set are not that long. Would you like to contribute? I think a good implementation on this will be very meaningful.
applications/Colossal-LLaMA-2/colossal_llama2/dataset/spliced_and_tokenized_dataset.py
